app/assets/javascripts/Components/grade_entry_form_chart.jsx

```diff
@@ -68,56 +65,37 @@ export class GradeEntryFormChart extends React.Component {
 
   render() {
     return (
-      <React.Fragment>
-        <h2>
+      <AssessmentChart
+        course_id={this.props.course_id}
+        assessment_id={this.props.assessment_id}
+        assessment_header_content={
           <a
             href={Routes.edit_course_grade_entry_form_path(
               this.props.course_id,
               this.props.assessment_id
             )}
           >
-            {this.state.info_data.name}
+            {this.props.show_chart_header ? this.state.summary.name : ""}
           </a>
-        </h2>
-        <div className="flex-row">
-          <div>
-            <Bar
-              data={this.state.distribution_data}
-              options={this.state.distribution_data.options}
-              width="500"
-              height="450"
+        }
+        summary={this.state.summary}
+        assessment_data={this.state.grade_entry_form_distribution.data}
+        additional_assessment_stats={
+          <React.Fragment>
+            <span className="summary-stats-label">{I18n.t("attributes.date")}</span>
+            <span>{this.state.summary.date}</span>
+            <span className="summary-stats-label">{I18n.t("num_entries")}</span>
+            <FractionStat
+              numerator={this.state.summary.num_entries}
+              denominator={this.state.summary.groupings_size}
             />
-          </div>
-
-          <div className="flex-row-expand">
-            <p>
-              {this.state.info_data.date &&
-                I18n.t("attributes.date") + ": " + this.state.info_data.date}
-            </p>
-
-            <div className="grid-2-col">
-              <span>{I18n.t("average")}</span>
-              <span>{(this.state.info_data.average || 0).toFixed(2)}%</span>
-              <span>{I18n.t("median")}</span>
-              <span>{(this.state.info_data.median || 0).toFixed(2)}%</span>
-              <span>{I18n.t("num_entries")}</span>
-              <span>{this.state.info_data.num_entries}</span>
-              <span>{I18n.t("num_failed")}</span>
-              <span>{this.state.info_data.num_fails}</span>
-              <span>{I18n.t("num_zeros")}</span>
-              <span>{this.state.info_data.num_zeros}</span>
-            </div>
-          </div>
-        </div>
-
-        <h3>{I18n.t("grade_entry_forms.grade_entry_item_distribution")}</h3>
-        <Bar
-          data={this.state.column_data}
-          options={this.state.column_data.options}
-          width="400"
-          height="350"
-        />
-      </React.Fragment>
+          </React.Fragment>
+        }
+        secondary_grade_distribution_title={I18n.t(
+          "grade_entry_forms.grade_entry_item_distribution"
+        )}
+        secondary_grade_distribution_data={this.state.column_grade_distribution.data}
```

Comment:
@anoverga17 great work! I have to jump into a meeting so I'm giving only two high-level comments in this review. The first is here: grade entry form *columns* are analogous to assignment *criteria*, and so should be displayed in the same way. In other words you should compute the same stats for each column like you did for each criterion, and display the column grade information by passing the data to the `criteria` prop rather than as "`secondary_grade_distribution`". You can use props to customize the title/labels of "criteria" versus "columns".

In terms of the Dashboard: I think the column grades are being shown in the dashboard graphs, which is maybe why you chose to use the "`secondary_grade_distribution`" props for that. To preserve the current UI, maybe you can use a `showCriteriaSummary` prop or something.

(As a note, grade entry forms *also* assign graders to students, so the stats summary could display grader information analogous to that of assignments. But in the interest of time you don't need to add that in this PR.)